### PR TITLE
Added schema 2.0

### PR DESCRIPTION
**Pull request by samuellye98**
Changed $template from string to an object with fields: name, type, url

### CODE REVIEW

File: schema/2.0/changelog [hunk 1]
```diff
@@ -0,0 +1,8 @@
+changelog
+
+Added qualificationLevel field
+Added attainmentDate date-time string
+Added fieldOfStudy field
+Added field to transcript: languageMedium
+Added fields to recipient: studentId, nric, fin
```
Comment (yehjxraymond) [inline schema/2.0/changelog]: These were not in the change log right?

File: schema/2.0/schema.test.js [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+        object: {
+          hello: "world"
+        }
+      }
+    };
+    const signedDocument = issueDocument(data, schema);
+    const valid = validateSchema(signedDocument);
+    assert(valid);
+  });
+
```
Comment (yehjxraymond) [inline schema/2.0/schema.test.js]: Can you add a test to see what happens when your $template is an empty object? It should fail. 

$template: {}

Comment (yehjxraymond) [reply schema/2.0/schema.test.js]: Can put that is a describe block describe("schema 2.0 specific additions", () => {

**Comment by yehjxraymond** [comment]
You need to upgrade the default schema version to 2.0

File: schema/2.0/schema.json [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    "id": {
+      "type": "string"
+    },
+    "$template": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "type": {
```
Comment (yehjxraymond) [inline schema/2.0/schema.json]: Oh you can use an enum here. See: https://github.com/TradeTrust/tradetrust-schema/blob/master/schema/1.0/schema.json#L20

Also add a test to see if the field is something else, it should fail.

File: schema/2.0/schema.test.js [hunk 4]
```diff
@@ -0,0 +1,545 @@
...
+        name: "Recipient Name"
+      }
+    };
+
+    const signedDocument = issueDocument(data, schema);
+    const valid = validateSchema(signedDocument);
+    assert(valid);
+  });
+
+  describe("schema 2.0specific additions", () => {
```
Comment (yehjxraymond) [inline schema/2.0/schema.test.js]:
```suggestion
  describe("schema 2.0 specific additions", () => {
```
